apps/content/migrator/migrations/6/index.js

```diff
@@ -0,0 +1,60 @@
+const mongoose = require('mongoose')
+const { appConfiguration } = require('../../../config')
+const TENANT = process.env.TENANT || '0'
+
+const Configuration = mongoose.model('Configuration')
+
+/**
+ * update site configuration
+ */
+
+/**
+ * check if app-configuration doesn't have themeStylesUrl propery
+ */
+async function check () {
+  const hasMissingThemeConfig = await Configuration.countDocuments({ key: appConfiguration, 'metadata.themeStylesUrl': {$exists: false} })
+  return !hasMissingThemeConfig
+}
+
+/**
+ * migrate relevant db rows to fit the new upgrade
+ */
+function migrate () {
+  console.log('start appConfiguration.metadata.themeStylesUrl migration')
+  return Configuration.aggregate([
+    { $match: { 'metadata.themeStylesUrl': {$exists: false} } },
+    { $limit: 20 },
+    { $project: { _id: 1 } }
+  ])
+    .then(async (list) => {
+      if (!list || !list.length) {
+        console.log('No more rows to update.')
+        return true
+      }
```

Comment:
I'm trying to think of a way to make this block a more story telling.. 
one option:

```
function updateList(list) {
      let row
      console.log('start migration of ' + list.length + ' items')
      for (let i in list) {
        row = list[i]
        console.log('update appConfig: ', row._id)
        return Configuration.collection.updateOne({ _id: row._id },
          {
            $set: { 'metadata.themeStylesUrl': '' }
          })
      }
      }
```
and from the handler:

```
.then(async (list) =>  !list?.length ? true : (await updateList(list) && migrate())
```

Reply:
It should be async, lazy, and non-blocking.. that's why it's longer.